src/core/gui/TextEditor.cpp

```diff
@@ -28,20 +25,112 @@
 
 class UndoAction;
 
+/** GtkTextBuffer helper functions **/
+static auto getIteratorAtCursor(GtkTextBuffer* buffer) -> GtkTextIter {
+    GtkTextIter cursorIter = {nullptr};
+    gtk_text_buffer_get_iter_at_mark(buffer, &cursorIter, gtk_text_buffer_get_insert(buffer));
+    return cursorIter;
+}
+
+/**
+ * @brief Compute the byte offset of an iterator in the GtkTextBuffer
+ *
+ * NB: This is much faster than relying on g_utf8_offset_to_pointer
+ */
+static auto getByteOffsetOfIterator(GtkTextIter it) -> int {
+    // Bytes from beginning of line to iterator
+    int pos = gtk_text_iter_get_line_index(&it);
+    gtk_text_iter_set_line_index(&it, 0);
+    // Count bytes of previous lines
+    while (gtk_text_iter_backward_line(&it)) {
+        pos += gtk_text_iter_get_bytes_in_line(&it);
+    }
+    return pos;
+}
+
+static auto getByteOffsetOfCursor(GtkTextBuffer* buffer) -> int {
+    return getByteOffsetOfIterator(getIteratorAtCursor(buffer));
+}
+
+/**
+ * @brief Get an iterator at the prescribed byte index.
+ *
+ * NB: This is much faster than relying on g_utf8_pointer_to_offset for long texts
+ */
+static auto getIteratorAtByteOffset(GtkTextBuffer* buf, int byteIndex) {
+    assert(byteIndex >= 0);
+    GtkTextIter it = {nullptr};
+    gtk_text_buffer_get_start_iter(buf, &it);
+
+    // Fast forward to the beginning of the line containing our target destination
+    for (int linelength = gtk_text_iter_get_bytes_in_line(&it);
+         linelength <= byteIndex && gtk_text_iter_forward_line(&it);
+         byteIndex -= std::exchange(linelength, gtk_text_iter_get_bytes_in_line(&it))) {}
+
+    // If byteIndex is past the line end (so past the buffer end), this will not change it
+    gtk_text_iter_set_line_index(&it, byteIndex);
```

Comment:
Actually this has some issues. When moving to an empty line I get the following warning (with level `Gtk-CRITICAL`):

``` term
(xournalpp:50699): Gtk-CRITICAL **: 15:05:32.081: gtk_text_iter_set_line_index: assertion 'byte_on_line <= bytes_in_line' failed
```
So it looks better to avoid this.

Reply:
Ok, this is fixed now.